test/blackbox/common/BlackboxTestsNetworkConf.cpp

```diff
@@ -197,3 +192,27 @@ TEST(BlackBox, SubGetListeningLocators)
         }
     }
 }
+
+TEST(BlackBox, PubGetSendingLocators)
+{
+    PubSubWriter<HelloWorldType> writer(TEST_TOPIC_NAME);
+
+    constexpr uint32_t port = 31337u;
+
+    std::shared_ptr<UDPv4TransportDescriptor> descriptor = std::make_shared<UDPv4TransportDescriptor>();
+    descriptor->m_output_udp_socket = static_cast<uint16_t>(port);
+
+    writer.disable_builtin_transport().
+        add_user_transport_to_pparams(descriptor).
+        init();
+
+    ASSERT_TRUE(writer.isInitialized());
+
+    LocatorList_t locators;
+    writer.get_native_writer().get_sending_locators(locators);
+
+    EXPECT_FALSE(locators.empty());
+    Locator_t locator = *(locators.begin());
+    EXPECT_EQ(locator.port, port);
+    EXPECT_EQ(locator.kind, LOCATOR_KIND_UDPv4);
```

Comment:
Should we check that there is only one locator returned? Or could there be more besides the one the user configured?

Reply:
More than one sending locator may be created to allow multicast to be sent through all interfaces. Nevertheless, I will add another test with several interfaces on the whitelist. In that case, we know the exact list of locators that should be returned.

Reply:
Done in ea4b5eb